debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerStreamingChangeEventSource.java

```diff
@@ -83,98 +99,190 @@ public void execute(ChangeEventSourceContext context) throws InterruptedExceptio
                 final Lsn fromLsn = lastProcessedLsn.isAvailable() ? connection.incrementLsn(lastProcessedLsn)
                         : lastProcessedLsn;
 
-                connection.getChangesForTables(tables, fromLsn, currentMaxLsn, resultSets -> {
-                    final int tableCount = resultSets.length;
-                    final ChangeTable[] changeTables = new ChangeTable[tableCount];
-
-                    for (int i = 0; i < tableCount; i++) {
-                        changeTables[i] = new ChangeTable(tables[i], resultSets[i]);
-                        changeTables[i].next();
+                while (!schemaChangeCheckpoints.isEmpty()) {
+                    migrateTable(schemaChangeCheckpoints);
+                }
+                if (!connection.listOfNewChangeTables(fromLsn, currentMaxLsn).isEmpty()) {
+                    final ChangeTable[] tables = getCdcTablesToQuery();
+                    tablesSlot.set(tables);
+                    for (ChangeTable table: tables) {
+                        if (table.getStartLsn().isBetween(fromLsn, currentMaxLsn)) {
+                            LOGGER.info("Schema will be changed for {}", table);
+                            schemaChangeCheckpoints.add(table);
+                        }
                     }
+                }
+                try {
+                    connection.getChangesForTables(tablesSlot.get(), fromLsn, currentMaxLsn, resultSets -> {
+
+                        final int tableCount = resultSets.length;
+                        final ChangeTablePointer[] changeTables = new ChangeTablePointer[tableCount];
+                        final ChangeTable[] tables = tablesSlot.get();
 
-                    for (;;) {
-                        ChangeTable tableSmallestLsn = null;
                         for (int i = 0; i < tableCount; i++) {
-                            final ChangeTable changeTable = changeTables[i];
-                            if (changeTable.isCompleted()) {
+                            changeTables[i] = new ChangeTablePointer(tables[i], resultSets[i]);
+                            changeTables[i].next();
+                        }
+
+                        for (;;) {
+                            ChangeTablePointer tableSmallestLsn = null;
```

Comment:
`tableSmallestLsn` suggests it's `Lsn` object, which is misleading. `tableWithSmallestLsn`?

Reply:
Renamed